comet/comet_test.go

```diff
@@ -116,13 +125,24 @@ func TestStatus(t *testing.T) {
 }
 
 func TestBroadcastTx(t *testing.T) {
-	chainID := "0"
-	app := testapp.NewTest(t, chainID)
+	app := testapp.NewTest(t, chainID.String())
+	_, err := app.InitChain(context.Background(), &abcitypes.RequestInitChain{
+		ChainId: chainID.String(),
+		AppStateBytes: func() []byte {
+			appStateBytes, err := json.Marshal(testapp.MakeGenesisAppState(t, app))
+			require.NoError(t, err)
+			return appStateBytes
+		}(),
+	})
+	require.NoError(t, err)
+
+	ctx := app.GetContext(true)
+	sk, _, acc := app.TestAccount(ctx)
```

Comment:
This is a common pattern across every test that needs to sign transactions with testapp.ToTx. We need to call InitChain before we fetch the application context, and then we get our pre-funded test account with app.TestAccount.

It'd probably be good to encapsulate this setup logic in a helper function (perhaps inside of testapp.NewTest itself), so we can avoid repeating it across each test.

Reply:
That sounds great! Should make the test much easier to review. The diff is a bit scary atm :)

Reply:
Agreed, that was my intention with this comment. I realized the diff was a bit scary 😆. Will re-ping when RFR.

Reply:
@joshklop I thought for a while about the best way to simplify the changes here, and settled on refactoring `testapp.NewTest` to encapsulate this repeated setup logic for us. However, there are some tests that call `genesis.Commit` and some that don't -- and there are some tests that need access to a `blockStore` and `ethstatedb`, while others don't.

My idea for a solution here is to create a struct like this:
```go
// TestApp wraps a boilerplate Cosmos SDK application with auxiliary data structures for testing.
type TestApp struct {
	*App
	Ctx        sdk.Context
	ChainID    monomer.ChainID
	Genesis    *genesis.Genesis
	BlockStore *localdb.DB
	EthStateDB state.Database
	PrivKey    *secp256k1.PrivKey
	Account    sdk.AccountI
}
```
and have `NewTest` call `InitChain` for us (and optionally `genesis.Commit`) then return this structure. That way the application is already initialized for us in every test, and each test has access to these auxiliary components if they need them. I also made a few other changes to helper functions to shrink the footprint of this diff, though please feel free to lmk if there's more I can do to make review easier.